projects/plugins/jetpack/modules/sharedaddy/sharing.php

```diff
@@ -29,7 +29,6 @@ public function __construct() {
 		require_once WP_SHARING_PLUGIN_DIR . 'sharing-service.php';
 
 		add_action( 'admin_init', array( $this, 'admin_init' ) );
-		add_action( 'admin_menu', array( $this, 'subscription_menu' ) );
```

Comment:
Things are a bit more complex here. While the Likes, Comment Likes, and Publicize features rely on a WordPress.com connection, the Sharing feature can be used without a connection to WordPress.com. That means that we need to allow folks who have not connected their site to WordPress.com to configure their sharing buttons in wp-admin.

To reproduce this use-case, create a new self-hosted test site and enable [offline mode](https://jetpack.com/support/offline-mode/) (if you use Jurassic Ninja, go to Settings > Jetpack Constants to enable that option, if not you can add `define( 'JETPACK_DEV_DEBUG', true );` to your site's `wp-config.php` file). You'll then want to go to Appearance > Themes and install / activate a classic theme (block-based themes do not need a sharing buttons configuration screen anymore, they can go to the site editor and add a sharing block instead). Then go to Jetpack > Settings > Sharing and enable sharing buttons. You'll see the link to wp-admin sharing settings appear at the bottom of the card as soon as the module is active. That link is currently broken since this PR, since we're not registering the page anymore.

I would recommend that we revert that part of your PR, while adding conditions so the menu only gets registered in the right scenario. Something like this:

```diff
diff --git a/projects/plugins/jetpack/modules/sharedaddy/sharing.php b/projects/plugins/jetpack/modules/sharedaddy/sharing.php
index 67bca06ac5..d941482a6a 100644
--- a/projects/plugins/jetpack/modules/sharedaddy/sharing.php
+++ b/projects/plugins/jetpack/modules/sharedaddy/sharing.php
@@ -10,6 +10,7 @@
 use Automattic\Jetpack\Assets;
 use Automattic\Jetpack\Redirect;
 use Automattic\Jetpack\Status;
+use Automattic\Jetpack\Status\Host;
 
 if ( ! defined( 'WP_SHARING_PLUGIN_URL' ) ) {
 	define( 'WP_SHARING_PLUGIN_URL', plugin_dir_url( __FILE__ ) );
@@ -29,6 +30,7 @@ class Sharing_Admin {
 		require_once WP_SHARING_PLUGIN_DIR . 'sharing-service.php';
 
 		add_action( 'admin_init', array( $this, 'admin_init' ) );
+		add_action( 'admin_menu', array( $this, 'subscription_menu' ) );
 
 		// Insert our CSS and JS
 		add_action( 'load-settings_page_sharing', array( $this, 'sharing_head' ) );
@@ -122,23 +124,22 @@ class Sharing_Admin {
 
 	/**
 	 * Register Sharing settings menu page.
+	 *
+	 * We only need to register it on self-hosted sites, in offline mode.
 	 */
 	public function subscription_menu() {
-		if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
-			$active = Jetpack::get_active_modules();
-			if (
-				! in_array( 'publicize', $active, true )
-				&& ! current_user_can( 'manage_options' )
-			) {
-				return;
-			}
+		if (
+			( new Host() )->is_wpcom_platform()
+			|| ! ( new Status() )->is_offline_mode()
+		) {
+			return;
 		}
 
 		add_submenu_page(
 			'options-general.php',
 			__( 'Sharing Settings', 'jetpack' ),
 			__( 'Sharing', 'jetpack' ),
-			'publish_posts',
+			'manage_options',
 			'sharing',
 			array( $this, 'wrapper_admin_page' )
 		);
@@ -729,7 +730,7 @@ class Sharing_Admin {
 			new Share_Reddit( 'reddit', array() ),
 		);
 		?>
-		
+
 		<div class="share_manage_options">
 			<br class="clearing" />
 			<h2><?php esc_html_e( 'Sharing Buttons', 'jetpack' ); ?></h2>

```

Reply:
I'll revert this part soon

Reply:
Thanks for the review @jeherve!

Reply:
Proposing https://github.com/Automattic/jetpack/pull/35810 to address this issue 🙂